src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -555,6 +555,7 @@ private async Task KillCursorsAsync(CancellationToken cancellationToken)
         public bool MoveNext(CancellationToken cancellationToken)
         {
             ThrowIfDisposed();
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
NOTE: Not 100% sure that this change is required.
this change doesn't protect us from the situation when another thread calls `cancellation.Cancel` in the time when this line is already passed. But at least we cover some of the possible situations. 

The benefit which we have with this validation that if the cancellationToken is canceled before this line (so, an OperationCanceledException exception is triggered in this line), we won't call operation at all, so from the stacktrace point of view, it can be more expected from the user.

NOTE2: But we will throw the OperationCanceledException error in any case, even if the cancellation token has been canceled after the line `558`. But in this case, it will happen in the `GetChannel` method inside `GetNextBatch`

Reply:
I don't think this change is *required*.

But I do think it's a good idea. Let's keep it.